packages/cli/src/controllers/passwordReset.controller.ts

```diff
@@ -231,7 +231,7 @@ export class PasswordResetController {
 
 		const user = await this.userService.findOne({
 			where: { id: decodedToken.sub },
-			relations: ['authIdentities'],
+			relations: ['authIdentities', 'globalRole'],
```

Comment:
We really need better sub-classing of DB entities in the ORM code. So, instead of guessing if `user.globalRole` exists or not, we have have separate types with and without the `globalRole` property. That way issues like these could be caught during compile time.

Reply:
Can we just create a new type from User where the role is required (UserWithRole) and use in the functions where we need the role. Then developers will use a type guard function to change narrow the type from User to UserWithRole?

Reply:
friendly ping @netroy